pkg/otlp/config.go

```diff
@@ -163,7 +164,15 @@ func fromExperimentalConfig(cfg config.Config) (PipelineConfig, error) {
 // IsEnabled checks if OTLP pipeline is enabled in a given config.
 func IsEnabled(cfg config.Config) bool {
 	// TODO (AP-1267): Check stable config too
-	return isSetExperimental(cfg)
+	isSet := isSetExperimental(cfg)
+	// NOTE: it is perhaps not ideal to set this here, but:
+	// * we can't import otlp in inventories, because that triggers import loop
+	// * we don't want to use raw config in inventories, because the rules for
+	//   evaluating are quite complex and will change with stable config
+	// * this is called in cmd/agent/app/run.go before inventories, so the correct
+	//   value will be set before inventory metadata is used first time
+	inventories.SetAgentMetadata(inventories.AgentOTLPEnabled, isSet)
```

Comment:
Would it be better to call this from `cmd/agent/app/run.go` after calling `otlp.IsEnabled`? It's a bit unexpected that a getter function like this changes global state.

Reply:
➕, I also like that approach better

Reply:
I don't particularly like adding more code to that file, but I guess you're right that in this case it will probably be the best place to call this.